src/blob_storage.cc

```diff
@@ -45,14 +77,27 @@ void BlobStorage::ExportBlobFiles(
 void BlobStorage::AddBlobFile(std::shared_ptr<BlobFileMeta>& file) {
   MutexLock l(&mutex_);
   files_.emplace(std::make_pair(file->file_number(), file));
+  blob_ranges_.emplace(std::make_pair(file->smallest_key(), file));
```

Comment:
exclude files with empty smallest key?

Reply:
Nope. When DeleteBlobFilesInRange with range(nullptr, nullptr), we should delete these files as well.

Reply:
got it.